### PR TITLE
update java samples

### PR DESCRIPTION
**Pull request by wanjunlei**
Signed-off-by: wanjunlei <wanjunlei@kubesphere.io>

### CODE REVIEW

File: functions/knative/hello-world-java/README.md [hunk 1]
```diff
@@ -1,36 +1,132 @@
...
-    runtime: knative # default to knative
-    template:
-      containers:
-        - name: function # DO NOT change this
-          imagePullPolicy: IfNotPresent 
+## Prerequisites
+
+### OpenFunction
+
+You can refer to the [Installation Guide](https://github.com/OpenFunction/OpenFunction#install-openfunction) to setup OpenFunction.
```
Comment (benjaminhuo) [inline functions/knative/hello-world-java/README.md:7]: installation guide is now https://openfunction.dev/docs/getting-started/installation/

File: functions/knative/hello-world-java/README.md [hunk 2]
```diff
@@ -1,36 +1,132 @@
...
+   Hello, OpenFunction!
+   ```
+   
+   There is also an alternative way to trigger the function via the access address provided by the Knative Services:
+
+    ```shell
+    kubectl get ksvc
+     
+    NAME                       URL                                                            LATESTCREATED                   LATESTREADY                     READY   REASON
+    serving-q6wdp-ksvc-wk6mv   http://serving-q6wdp-ksvc-wk6mv.default.<external-ip>.sslip.io   serving-q6wdp-ksvc-wk6mv-v100   serving-q6wdp-ksvc-wk6mv-v100   True
```
Comment (benjaminhuo) [inline functions/knative/hello-world-java/README.md:99]: We now have function gateway, the old access method is outdated. https://openfunction.dev/docs/concepts/networking/function-entrypoints/

File: functions/knative/hello-world-java/README.md [hunk 3]
```diff
@@ -1,36 +1,132 @@
...
-      url: "https://github.com/OpenFunction/samples.git"
-      sourceSubPath: "functions/knative/hello-world-java"
-      revision: "main"
-  serving:
-    runtime: knative # default to knative
-    template:
-      containers:
-        - name: function # DO NOT change this
-          imagePullPolicy: IfNotPresent 
+## Prerequisites
```
Comment (benjaminhuo) [inline functions/knative/hello-world-java/README.md:3]: We need to put java function sample links to:
https://openfunction.dev/docs/getting-started/quickstarts/sync-functions/
https://openfunction.dev/docs/getting-started/quickstarts/async-functions/
https://openfunction.dev/docs/concepts/function_signatures/

**Comment by wanjunlei** [comment]
Close it first, and reopen after fixed.

**Pull request closed without merging**